lang/c++/CMakeLists.txt

```diff
@@ -81,11 +78,8 @@ if (AVRO_ADD_PROTECTOR_FLAGS)
 endif ()
 endif ()
 
-if (AVRO_BUILD_TESTS OR AVRO_BUILD_EXECUTABLES)
-    find_package (Boost 1.38 REQUIRED
-        COMPONENTS filesystem iostreams program_options system)
-else ()
-    find_package (Boost 1.38 REQUIRED COMPONENTS iostreams)
+if (AVRO_BUILD_TESTS OR AVRO_USE_BOOST)
+    find_package (Boost 1.38 REQUIRED COMPONENTS system)
```

Comment:
@thiru-mg I'm not sure if `boost/system` is required because I haven't found any header inclusion from all source files. Should we remove it as well?